server/crawlers/redditCrawler.js

```diff
@@ -0,0 +1,43 @@
+const snoowrap = require("snoowrap");
+
+async function redditCrawler(companyName) {
+  const scraper = new snoowrap({
+    userAgent: process.env.Reddit_userAgent,
+    clientId: process.env.Reddit_clientId,
+    clientSecret: process.env.Reddit_clientSecret,
+    refreshToken: process.env.Reddit_refreshToken,
+  });
+
+  const subreddit = await scraper.getSubreddit("all");
+  const topPosts = await subreddit.search({
+    query: companyName,
+    time: "all",
+    sort: "relevance",
+  });
+
+  let data = [];
+
+  topPosts.forEach((post) => {
+    // console.log(post);
+    // If a post's title doesn't contain the specified company name, or it doesn't have any content, just ignore it.
+    if (post.title.includes(companyName) && post.selftext) {
+      // If a post doesn't have a thumbnail picture, just use Reddit's logo.
+      if (post.thumbnail.substr(0, 8) !== "https://") {
+        post.thumbnail = "/imgs/reddit_icon.png";
```

Comment:
if it doesn't have a thumbnail, i would recommend save this as NULL rather than a hardcoded image allowing for better scalability in the future. for example, if you want to turn this into a user icon instead, or a diff image. instead, on the frontend, you can determine how you want to display it based off of its null value

Reply:
Makes sense, but, it would require more code in the FE. 
For example, when FE gets a mention and finds that its image is `null`, there has to be several `if-else` to check what platform the mention belongs to, such as
`if(mentions.playform == "Twitter")
...
else if (mentions.playform == "Reddit")
...
else if (mentions.playform == "Facebook")
...`
But if we hardcode the path in this scraper file, even if we decide to display a different image in the future, we can just do `if(mention.image.substr(0, 8) != "https://") {
// show another picture
}`
It seems that this way is a little simpler?

Reply:
Done for both the scraper files and the main page file.